doc/examples/nn_deformable_registration_2d.py

```diff
@@ -0,0 +1,272 @@
+"""
+
+import numpy as np
+import scipy.ndimage
+import math
+from dipy.nn.registration import FCN2d
+from dipy.nn.metrics import normalized_cross_correlation_loss
+from distutils.version import LooseVersion
+from dipy.utils.optpkg import optional_package
+plt, _, _ = optional_package("matplotlib.pyplot")
+tf, have_tf, _ = optional_package('tensorflow')
+if have_tf:
+    if LooseVersion(tf.__version__) < LooseVersion('2.0.0'):
+        raise ImportError('Please upgrade to TensorFlow 2+')
+
+"""
+To get started, we will load the MNIST dataset into the memory and filter 
+the training and testing sets to keep just one class of digits. We can 
+change the ``label`` variable to select a different class.
+"""
+
+# Load data
+(x_train, y_train), (x_test, y_test) = tf.keras.datasets.mnist.load_data()
+
+# Discard other digits.
+label = 7  # Which digit images to keep.
+x_train = x_train[y_train == label].copy()  # shape (n_train, 28, 28)
+x_test = x_test[y_test == label].copy()  # shape (n_test, 28, 28)
+
+# We'll also resize the images to (32, 32), just to keep it a power of 2.
+x_train = scipy.ndimage.zoom(x_train, (1, 32/28, 32/28))
+x_test = scipy.ndimage.zoom(x_test, (1, 32/28, 32/28))
+
+"""
+Let's select our static image randomly from the test set. We'll also select 
+few images (``x_sample``) for visualizing the results.
+"""
+
+idx = np.random.randint(x_test.shape[0])
+static = x_test[idx].copy()
+
+num_samples = 2  # Number of sample images to visualize.
+# Sample images to show results.
+idxs = np.random.choice(x_test.shape[0], replace=False,
+                        size=num_samples)
+x_sample = x_test[idxs].copy()  # shape (num_samples, 32, 32)
+
+"""
+Implement a data loader that fetches and preprocesses batches of images for 
+real-time data feeding to our model. 
+"""
+
+
+class DataLoader(tf.keras.utils.Sequence):
+    def __init__(self, x, static, batch_size=8, shuffle=False):
+        self.x = x
+        self.static = static
+        self.batch_size = batch_size
+        self.shuffle = shuffle
+
+        if self.shuffle:
+            np.random.shuffle(self.x)
+
+    def __len__(self):
+        return math.ceil(len(self.x) / self.batch_size)
+
+    def __getitem__(self, idx):
+        moving = self.x[idx * self.batch_size: (idx + 1) * self.batch_size]
+        moving = moving[..., np.newaxis]
+        static = self.static[np.newaxis, ..., np.newaxis]
+        static = np.repeat(static, repeats=moving.shape[0], axis=0)
+
+        # Rescale to [0, 1].
+        moving = moving.astype(np.float32)  # (N, 32, 32, 1)
+        static = static.astype(np.float32)  # (N, 32, 32, 1)
+        moving = moving / 255.0
+        static = static / 255.0
+
+        return {'moving': moving, 'static': static}, static
+
+    def on_epoch_end(self):
+        if self.shuffle:
+            np.random.shuffle(self.x)
+
+
+"""
+Let's define some hyperparameters used for training the network.
+"""
+
+batch_size = 32
+epochs = 100
```

Comment:
It looks like performance converges after much fewer epochs. Maybe set this to 25 for the purpose of the example and then add a comment that this could be increased to a larger number for more accurate results?